legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/project/maven/MavenProjectStructure.java

```diff
@@ -275,10 +276,16 @@ public void addJacksonDependency(Consumer<Dependency> dependencyConsumer, boolea
 
     public static Stream<Dependency> projectDependencyToMavenDependencies(ProjectDependency projectDependency, BiFunction<String, VersionId, ProjectFileAccessProvider.FileAccessContext> versionFileAccessContextProvider, Collection<? extends ArtifactType> artifactTypes, boolean setVersion)
     {
+        String versionString = setVersion ? projectDependency.getVersionId().toVersionIdString() : null;
+        if (!ProjectDependency.isLegacyProjectDependency(projectDependency))
+        {
+            List<String> mavenCoordinates = ProjectDependency.getMavenCoordinatesFromProjectDependency(projectDependency.getProjectId());
+            List<ArtifactType> artifactTypeList = ((artifactTypes == null) || artifactTypes.isEmpty()) ? Arrays.asList(ArtifactType.entities,ArtifactType.versioned_entities) : artifactTypes.stream().collect(Collectors.toList());
+            return artifactTypeList.stream().map(artifactType -> newMavenDependency(mavenCoordinates.get(0), mavenCoordinates.get(1) + "-" + artifactType, versionString));
```

Comment:
The default set of artifact types should be a static field. E.g.,

`private static final ImmutableList<ArtifactType> DEFAULT_ARTIFACT_TYPES = Lists.immutable.with(ArtifactType.entities, ArtifactType.versioned_entities);`

Then this bit of code should be something like this:
```suggestion
            Collection<? extends ArtifactType> resolvedArtifactTypes = ((artifactTypes == null) || artifactTypes.isEmpty()) ? DEFAULT_ARTIFACT_TYPES : artifactTypes;
            return resolvedArtifactTypes.stream().map(artifactType -> newMavenDependency(mavenCoordinates.get(0), mavenCoordinates.get(1) + "-" + artifactType, versionString));
```